finetuner/tuner/dataset/__init__.py

```diff
@@ -1,8 +1,158 @@
 import itertools
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 import numpy as np
 
 from ... import __default_tag_key__
+from ...helper import DocumentSequence
+
+
+class ClassDataset:
+    """ Dataset for enapsulating data where each item has a class label."""
+
+    def __init__(
+        self,
+        docs: DocumentSequence,
+        preprocess_fn: Optional[Callable[[Union[str, np.ndarray]], Any]] = None,
+    ):
+        """Create the :class:`ClassDataset` instance.
+
+        :param docs: The documents for the dataset. Each document is expected to have
+            - a content (only blob or text are accepted currently)
+            - a class label, saved under ``tags['finetuner']['label']``. This class
+              label should be an integer or a string
+        :param preprocess_fn: A pre-processing function. It should take as input the
+            content of an item in the dataset (currently only text of blob are
+            accepted).
+        """
+        self._docs = docs
+        self._preprocess_fn = preprocess_fn
+
+        self._tag_labels_dict: Dict[Union[str, int], int] = {}
+        self._labels: List[int] = []
+        max_label = 0
+
+        for doc in self._docs:
+            try:
+                tag = doc.tags[__default_tag_key__]['label']
+            except KeyError as e:
+                raise KeyError(
+                    'The tag ["finetuner"]["label"] was not found in a document.'
+                    ' When using ClassDataset all documents need this tag'
+                ) from e
+            label = self._tag_labels_dict.get(tag)
+
+            if label is None:
+                label = max_label
+                self._tag_labels_dict[tag] = label
+                max_label += 1
+
+            self._labels.append(label)
+
+    def __getitem__(self, ind: int) -> Tuple[Union[np.ndarray, str], int]:
+        """
+        Get the (preprocessed) content and label for the item at ``ind`` index in the
+        dataset.
+        """
+        content = self._docs[ind].content
+        label = self._labels[ind]
+
+        if self._preprocess_fn is not None:
```

Comment:
minor thing, I think `is not None` is not necessary, when writing boolean check, python did the job for us:
```python
if foo:
      # enter True branch

if not foo:
    # enter False branch
```